nomenclature/testing.py

```diff
@@ -52,15 +52,26 @@ def assert_valid_structure(
     Folder structure of `path`:
         - A `definitions` folder is required and must be a valid
         :class:`DataStructureDefinition`
+        - The `definitions` folder must contain sub-folder(s) to validate
         - If a `mappings` folder exists, it must be a valid :class:`RegionProcessor`
 
     """
     if not (path / definitions).is_dir():
         raise NotADirectoryError(
             f"Definitions directory not found: {path / definitions}"
         )
-    if dimensions is None:
+
+    if dimensions is None:  # if "dimensions" were not specified
         dimensions = [x.stem for x in (path / definitions).iterdir() if x.is_dir()]
+        if not dimensions:
+            raise FileNotFoundError(
+                f"`definitions` directory is empty: {path / definitions}"
+            )
+
+    if dimensions == []:  # if "dimensions" were specified as "[]"
+        raise FileNotFoundError(
+            f"Invalid dimensions: {dimensions}, there is nothing to validate."
```

Comment:
I think that this will be shown as 
> Invalid dimensions: , there is nothing to validate.
which is not quite helpful. Maybe better
> No dimensions to validate.